equistore-core/src/io/npy_header.rs

```diff
@@ -308,79 +256,269 @@ impl From<std::io::Error> for WriteHeaderError {
+
+                let name = self.parse_string()?;
+
+                self.skip_whitespaces();
+                self.expects(',')?;
+                self.skip_whitespaces();
+
+                let value = self.parse_string()?;
+
+                self.skip_whitespaces();
+                self.expects(')')?;
+                self.skip_whitespaces();
+
+                data_type.push((name, value));
+
+                if self.current() == ',' {
+                    self.advance();
+                    self.skip_whitespaces();
+                } else {
+                    self.expects(']')?;
+                    break;
+                }
+
+                if self.current() == ']' {
+                    self.advance();
+                    break;
+                }
+            }
+
+            return Ok(DataType::Compound(data_type));
+        } else {
+            return Err(ParseHeaderError::InvalidHeader(format!(
+                "expected a string or a list, got '{}'", self.current()
+            )));
+        }
+    }
+
+    fn parse_bool(&mut self) -> Result<bool, ParseHeaderError> {
+        if self.current() == 'T' {
+            self.advance();
+            self.expects('r')?;
+            self.expects('u')?;
+            self.expects('e')?;
+            return Ok(true);
+        } else if self.current() == 'F' {
+            self.advance();
+            self.expects('a')?;
+            self.expects('l')?;
+            self.expects('s')?;
+            self.expects('e')?;
+            return Ok(false);
+        } else {
+            return Err(ParseHeaderError::InvalidHeader(format!(
+                "expected a bool, got '{}'", self.current()
+            )));
+        }
+    }
+
+    fn parse_shape(&mut self) -> Result<Vec<usize>, ParseHeaderError> {
+        let mut shape = Vec::new();
+        self.expects('(')?;
+        loop {
+            self.skip_whitespaces();
+            shape.push(self.parse_integer()?);
+            self.skip_whitespaces();
+
+
+            if self.current() == ',' {
+                self.advance();
+                self.skip_whitespaces();
+            } else {
+                self.expects(')')?;
+                break;
+            }
+
+            if self.current() == ')' {
+                self.advance();
+                break;
+            }
+        }
+
+        return Ok(shape);
+    }
+
+    fn parse(&mut self) -> Result<Header, ParseHeaderError> {
+        let mut type_descriptor: Option<DataType> = None;
+        let mut fortran_order: Option<bool> = None;
+        let mut shape: Option<Vec<usize>> = None;
+
+        self.skip_whitespaces();
+        self.expects('{')?;
+        loop {
+            let key = self.parse_string()?;
+            self.skip_whitespaces();
+            self.expects(':')?;
+            self.skip_whitespaces();
+
+            if key == "descr" {
+                type_descriptor = Some(self.parse_data_type()?);
+            } else if key == "fortran_order" {
```

Comment:
What about "c_order"?

Reply:
the npy format stores `fortran_order: true` or `fortran_order: false` directly

Reply:
Ahh I see.